### DISCUSSION TITLE
Route visualization

### DISCUSSION THREAD

**Discussion by katepolak** [voxeltycoon/issues#538] [UI&UX]
I have been thinking about things that could be added to the route system, adding some more featured management UI would be great but is a bit out of scope right now. What could be done right now though, is showing the actual pathfinding route of the "route", the way to achieve that could simply be doing a game pathfinding call for the whole rout and than display it. This would be meant for the situations in which vehicles and stations on that route are highlighted.
While it could be a bit confusing since the vehicles might actually choose a different route, it would still be a nice thing to have so the player can easily see what the route is, and potentially see where it could be improved by reworking the infrastructure. This kinda ties into my conveyor highlight idea (#11) as the same visualization style and thus the same code could in theory be used, saving development time.

**Comment by Sliderpro** [comment]
I have similar suggestion. 
Sometimes, you have no idea why train doesn't move (not all people are good with signals). It would be cool if game would highlight the area of the signal when you point a cursor on it. Currently it just shows some arrows but if it would highlight the entire area which a particular signal sees, it would make life much easier!

**Comment by katepolak** [comment]
That's not similar at all I would say. Plus it's an duplicate, #380. Thanks for the idea though.
